.circleci/config.yml

```diff
@@ -181,6 +181,20 @@ jobs:
           paths:
             - ./build
 
+  # Check circular dependencies with cpp-dependencies tool
+  check_circular_deps:
+    <<: *defaults
+    steps:
+      - attach_workspace:
+          at: /hpx
+      - run:
+          name: Check circular dependencies with cpp-dependencies
+          command: |
+              ./cpp-dependencies --dir /hpx/source/libs /hpx/tools/cpp-dependencies/circular_deps.dot
+              if [[ $(wc -l /hpx/tools/cpp-dependencies/circular_deps.dot | awk '{print $1}') -gt 2 ]]; then exit 1; fi
+      - store_artifacts:
+          path: /hpx/tools/cpp-dependencies/circular_deps.dot
+
```

Comment:
The step needs to be added to the list of steps at the bottom of the config as well. I think it should be enough if this depends on the checkout step, and install depends on this step.